_studio/mfx_lib/vpp/src/mfx_vpp_hw.cpp

```diff
@@ -3689,6 +3689,20 @@ mfxStatus VideoVPPHW::MergeRuntimeParams(const DdiTask *pTask, MfxHwVideoProcess
             execParams->VideoSignalInfoOut.TransferMatrix = vsi->TransferMatrix;
      }
 
+    // Update ProcAmp parameters for output
+    mfxExtVPPProcAmp *procamp = reinterpret_cast<mfxExtVPPProcAmp *>(GetExtendedBuffer(outputSurf->Data.ExtParam,
```

Comment:
1. Do you have tests for it? (behavior or functional)
2. Why VPP::Reset() does not sufficient for this usage case?

Reply:
1. I plan to do functional testing using sample_vpp
2. Using VPP::Reset is not a good way if you have a night view with a bad camera that requires per frame HSBC adjustment.

Reply:
Was it a bug that HSBC changes weren't supported on fly? Or it's a new feature?

Reply:
While this doesn't use new API, this introduce undocumented change in how we use current API. Please put this code under feature define, like MFX_ENABLE_VPP_RUNTIME_HSBC and put this define in mfx_config.h under MFX_VERSION >= MFX_VERSION_NEXT guard.

Reply:
Oleg, could you, please, review and merge in this one: https://github.com/Intel-Media-SDK/MediaSDK/pull/146 before Roman will jump into changing config.h